### PR TITLE
Resolve issue 285 in parsing logic, KC.NO was being parsed has an alp…

### PR DESCRIPTION
**Pull request by pcharest2000**
Resolve issue 285 in parsing logic, KC.NO was being parsed has an alpha,  swapped the control flow treating the special case KC.NO and KC.XXXXX before alphas and nnumbers.   KC.NO was outputting n to OS when expexted behavior was to send nothing.

### CODE REVIEW

**Comment by kdb424** [comment]
There's a liner error that you can see in the test. Once that's resolved, looks clean.

**Review by klardotsh — APPROVED** [review]

**Comment by klardotsh** [comment]
I'm on mobile right now but I'll pull this on desktop in a bit and run the lint fixer, this is all autocorrectable.

**Comment by pcharest2000** [comment]
> I'm on mobile right now but I'll pull this on desktop in a bit and run the lint fixer, this is all autocorrectable.

I see the latest test failed again.... I don<t get it I use black to format the file!  Mybe I did push the latest file updating the pr....  let me try again

**Comment by klardotsh** [comment]
okay, this wasn't really your faul @pcharest2000. we use some overrides to `black` (that in retrospect I kinda wish I wouldn't have done, but too late now I guess unless we want to re-format the entire codebase... which I'm not opposed to), and `flake8` doesn't know how to properly auto-correct this one. so, these functions were using `"""` for docstrings where our `black` config was expecting `'''`. silly, and I'm sorry on behalf of past me.

I've force-pushed this branch as a single, squashed commit over `master` that includes my changes to appease the linter. once these new tests pass I'll merge.

thanks for the contribution @pcharest2000, and sorry for the tooling making your life harder than it needed to be - that's our fault!
